department-of-veterans-affairs/caseflow
# CASEFLOW-1121 | Substitute Appellant | Show Case Timeline Event

Connects CASEFLOW-1121

### Description
This builds out better support for non-task timeline events (such as decision date, NOD updates, and — specifically for this — substitution dates).

This renames/refactors a couple of existing functions to make clear that certain timeline events are coming from parts of the appeal object other than tasks.


### Acceptance Criteria
- [ ] Event is shown on Case Timeline if appeal has a substitution date (via an `appellantSubstitution` attached to it)

### Testing Plan
For new substitution event:
1. Put a `binding.pry` on line `112` of `spec/feature/queue/substitute_appellant/shared_setup.rb`
2. Run `substitute_appellant_spec.rb`
3. See the timeline event

For preserving existing capabilities:
1. Verify that "should sort tasks and nod date updates properly" test runs properly in `case_details_spec`


### User Facing Changes

| BEFORE|AFTER |
 ---|---
| ![Screen Shot 2021-05-13 at 10 07 15](https://user-images.githubusercontent.com/2581274/118137279-06e3dd00-b3d3-11eb-9ab7-4216644862bd.png) | ![Screen Shot 2021-05-12 at 18 23 50](https://user-images.githubusercontent.com/2581274/118132428-af8f3e00-b3cd-11eb-913f-86f6935d5e40.png) |
